tubesync/sync/filtering.py

```diff
@@ -127,20 +127,15 @@ def filter_max_cap(instance: Media):
     return False
 
 
-# If the source has a cut-off, check the upload date is within the allowed delta
-# TODO: days to keep should be compared to downloaded date, not published
+# If the source has a cut-off, check the download date is within the allowed delta
 def filter_source_cutoff(instance: Media):
-    if instance.source.delete_old_media and instance.source.days_to_keep > 0:
-        if not isinstance(instance.published, datetime):
-            # Media has no known published date or incomplete metadata
-            log.info(
-                f"Media: {instance.source} / {instance} has no published date, skipping"
-            )
-            return True
+    if instance.source.delete_old_media and instance.source.days_to_keep_date:
+        if not instance.downloaded or not isinstance(instance.download_date, datetime):
+            return False
 
-        delta = timezone.now() - timedelta(days=instance.source.days_to_keep)
-        if instance.published < delta:
-            # Media was published after the cutoff date, skip it
+        days_to_keep_age = instance.source.days_to_keep_date
+        if instance.download_date < days_to_keep_age:
```

Comment:
This can do a `datetime.datetime < False` comparison and raise a `TypeError` if `source.days_to_keep <= 0` I think? Just needs an `if days_to_keep_age and ...` check.

Reply:
Check the first of conditions.

Reply:
Oh right you evaluate it twice, fair enough. Reviewing code on a phone is hard.